src/hyper/src/connector.rs

```diff
@@ -18,19 +19,41 @@ use std::io::Read;
 use url::Url;
 
 #[derive(Clone, Debug)]
-pub struct SlackClientHyperConnector {
-    hyper_connector: Client<HttpsConnector<HttpConnector>>,
+pub struct SlackClientHyperConnector<H: Send + Sync + Clone + connect::Connect> {
+    hyper_connector: Client<H>,
 }
 
-impl SlackClientHyperConnector {
+pub type SlackClientHyperHttpsConnector =
+    SlackClientHyperConnector<HttpsConnector<HttpConnector>>;
+pub type SlackClientHyperProxyHttpsConnector =
+    SlackClientHyperConnector<ProxyConnector<HttpsConnector<HttpConnector>>>;
+
+impl SlackClientHyperHttpsConnector {
     pub fn new() -> Self {
         let https_connector = HttpsConnector::with_native_roots();
         let http_client = Client::builder().build::<_, hyper::Body>(https_connector);
         Self {
             hyper_connector: http_client,
         }
     }
+}
+
+impl SlackClientHyperProxyHttpsConnector {
+    pub fn with_proxy(url: &str) -> Self {
```

Comment:
It would be great to extract the code to configure proxy to the user side from library.
For example to be able to use authentication settings, etc, so the library would accept `Proxy` or `ProxyConnector` to be flexible for different environments and needs.
```
        let proxy_uri = "http://my-proxy:8080".parse().unwrap();
        let mut proxy = Proxy::new(Intercept::All, proxy_uri);
        proxy.set_authorization(Authorization::basic("John Doe", "Agent1234"));
```

Reply:
I'm thinking it is better to even have all possible options that users can specify anything that has `H: Send + Sync + Clone + connect::Connect` so you can use any kind of "connectors"/"proxies" and their options for Hyper if needed.

Reply:
Great, it looks now great. I'll do some changes and docs further. I see some inconsistencies I want to resolve before releasing - won't take too much.  Thanks for your effort!

Reply:
Thanks!

Reply:
@nhey 
https://slack-rust.abdolence.dev/hyper-connections-types.html
I made some slight internal changes, docs, and example for this. Also removed some stuff around it. One type that was used only in one place and your example was also removed, so please be aware :)